pytest-embedded/pytest_embedded/plugin.py

```diff
@@ -1252,7 +1253,35 @@ def _duplicate_items(items: t.List[_T]) -> t.List[_T]:
         return duplicates
 
     @pytest.hookimpl(hookwrapper=True, trylast=True)
-    def pytest_collection_modifyitems(self, items: t.List[Function]):
+    def pytest_collection_modifyitems(self, config: Config, items: t.List[Function]):
+        for item in items:
+            skip_marker = item.get_closest_marker('skip_if_soc')
+            if not skip_marker:
+                continue
+
+            target = config.getoption('--target', None)
+            if hasattr(item, 'callspec'):
+                target = item.callspec.params.get('target', None)
+            if target == 'auto' or not isinstance(target, str):
+                logging.warning(
+                    f"Test item '{item.originalname}': "
+                    "Ensure 'target' is included in the test @pytest.mark.parametrize when using 'skip_if_soc', "
+                    "or provide the --target argument when running tests (except 'auto' and multi-dut)."
+                )
+                continue
+            if '|' in target:
+                logging.warning("Multi-dut tests do not support the 'skip_if_soc' marker.")
+                continue
+
+            from esp_bool_parser import parse_bool_expr
+
+            stm = parse_bool_expr(skip_marker.args[0])
+            if not stm.get_value(target, ''):
+                from _pytest.mark import MARK_GEN
+
+                reason = f'Filtered by {skip_marker.args[0]}, for {target}.'
+                item.add_marker(MARK_GEN.skip(reason=reason))
```

Comment:
```suggestion
                reason = f'Filtered by {skip_marker.args[0]}, for {target}.'
                item.add_marker(pytest.mark.skip(reason=reason))
```
this is more like their "public api"?

Reply:
yea, much better